src/Columns/ColumnVariant.cpp

```diff
@@ -426,11 +426,30 @@ void ColumnVariant::insertData(const char *, size_t)
 }
 
 void ColumnVariant::insert(const Field & x)
+{
+    if (!tryInsert(x))
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Cannot insert field {} into column {}", toString(x), getName());
+}
+
+bool ColumnVariant::tryInsert(const DB::Field & x)
 {
     if (x.isNull())
+    {
         insertDefault();
-    else
-        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Cannot insert field {} to column {}", toString(x), getName());
+        return true;
+    }
+
+    for (size_t i = 0; i != variants.size(); ++i)
+    {
+        if (variants[i]->tryInsert(x))
```

Comment:
If it's possible to insert into several variants, shouldn't we choose the most specific? But it would work if Filed had specific numeric type inside. But since it's only (U)Int64, we will just insert to first sub-column with matching sign, probably also ok